### DISCUSSION TITLE
Can't resolve '@vercel/turbopack-next/internal/font/google/font' sporadically on `next dev --turbo`.

### DISCUSSION THREAD

**Discussion by lancejpollard** [vercel/next.js#61886] [Help]
### Summary

Keep getting this sporadically when I start my server, then I have to try and try again to restart it, until it finally works.

<img width="956" alt="Screenshot 2024-02-10 at 12 44 45 AM" src="https://github.com/vercel/next.js/assets/28718/b9459126-3952-44f6-9c95-5a6d9c9fc793">

### Additional information

```
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.2.0: Wed Nov 15 21:54:51 PST 2023; root:xnu-10002.61.3~2/RELEASE_ARM64_T6030
Binaries:
  Node: 20.10.0
  npm: 10.2.3
  Yarn: 1.22.19
  pnpm: 8.10.0
Relevant Packages:
  next: 14.1.1-canary.46 // Latest available version is detected (14.1.1-canary.46).
  eslint-config-next: 13.1.5
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

**Comment by Sayakie** [comment]
I faced same issue and resolved by installing latest version issued: `pnpm add next@latest`

**Reply by vrease** [reply]
Thanks Sayakie 🙏🏼

**Reply by Gold240sx** [reply]
Worked for me 04/15/25

**Reply by hrbust-xuzhi** [reply]
Worked for me 05/18/25

**Reply by AjiNurAji** [reply]
thanks sayakiee

**Reply by hasemon** [reply]
this works

**Reply by Sarawutchaiwong** [reply]
thanksXD

**Reply by mohamed-elhaissan** [reply]
what about npm ?

**Reply by calmliming** [reply]
thanks

**Reply by mohamed-hassaan3** [reply]
working well, thanks!

**Comment by ghost** [comment]
I am also experiencing the same issue with the error message: Module not found: Can't resolve '@vercel/turbopack-next/internal/font/google/font'. I have already updated all packages to the latest versions. Does anyone know how to resolve this? Thank you!

**Comment by shashicode** [comment]
I am still facing the same issue, tried reinstalling and adding the latest next, but nothing works, Has anyone found a fix yet?
these are building blocks of an application why this does not work :(

**Comment by niikkhilsharma** [comment]
Faced the same issue, I tried restarting the server and it worked for me.

**Comment by rsignavong** [comment]
I have the issue. How can I fix it?

**Reply by Nonellia** [reply]
Hey, did you solve it?

**Comment by devmev10** [comment]
Faced the same out of nowhere.
installing nextjs latest version and restarting vscode worked

> npm install next@latest

**Reply by devmev10** [reply]
it stopped working again. and now even restarting doesn't work.
Guess, have to find a permanent fix

**Reply by devmev10** [reply]
Found the issue - VPN!

on disabling, it worked.

**Reply by fegig** [reply]
I think it mostly has to do with networks. In my case, I had to turn on VPN before the issue went away.

**Reply by shalomspage** [reply]
I am facing the same issues. Do you know how I can fix it?

**Reply by shalomspage** [reply]
Restarting vs code worked for me.

**Reply by thiagocam-marleyspoon** [reply]
Thanks for the hint, I had a local proxy enabled and it worked after disabling it

**Reply by kanta1207** [reply]
Disconnecting the VPN made it work for me too. Thanks!

**Reply by hummxt** [reply]
Thanks dude.

**Reply by Gibson-Joseph** [reply]
Thanks @shalomspage

**Reply by WhiteSnowGirl** [reply]
thanks bro

**Comment by yusupsupriyadi** [comment]
I'm experiencing the same issue, but I temporarily turned off the turbopack feature when running the application on the deployment side by removing --turbopack in package.json

**Reply by hczs** [reply]
thanks!! It worked for me!!!

**Reply by mathanprasannakumar** [reply]
Thanks , it worked. Can any one know why such issue is happening , any leads would be welcomed.

**Reply by Kamaruddheen** [reply]
Thanks man. It worked

**Comment by ozgeonline** [comment]
Installing the latest versions may not solve the problem. Following the next.js documentation seems to have solved the problem for now. 
https://nextjs.org/docs/messages/built-in-next-font
npx @next/codemod built-in-next-font . 
I install it and it worked for me

**Reply by quyettranvu** [reply]
I tried that command but still not resolved, it kept showing sporadically even my tries seem to be very strict.

**Comment by quyettranvu** [comment]
I'm experiencing the same issue, I really love the behind architecture how Turbopack built but I really got mad because of this building error. Does anyone have any other suggestions? Thanks in advance!

**Reply by quyettranvu** [reply]
I hope that the developer team of Turbopack will check this soon

**Comment by sridhar2193** [comment]
I just removed --turbopack in next dev scripts in package.json. Now it is working. But need to debug why it is failing with turbopack

**Reply by daribock** [reply]
This solved the problem for me too. 🚀

**Comment by qyl27** [comment]
Same issue, can be reproduced in next `15.2.0`

**Comment by HenryOoO** [comment]
The same problem was solved by restarting the server

**Comment by stevenho0811** [comment]
delete `node_modules` folder, and reinstall, and delete `.next` folder solve my problem. and if you still facing same problem, try another network environment, and do it again.

**Comment by kybaloo** [comment]
It's the use of Turbopack, Adding this to `next.config.js` solved it for me:

```
const nextConfig = {
  experimental: {
    turbopack: false, // Désactive Turbopack
  },
};

module.exports = nextConfig;
```

**Comment by ukalpa** [comment]
Yeah, it seems to be related to Turbo, and the Google font Inter is used in the sample Turbo code.

If you don't want to remove `--turbo`, you can simply search for `import { Inter } from "next/font/google";` in your repository and delete the related code. That should resolve the issue.

https://github.com/vercel/turborepo/blob/main/examples/with-tailwind/apps/web/app/layout.tsx
<img width="560" alt="Screenshot 2025-03-20 at 07 45 22" src="https://github.com/user-attachments/assets/e69005ce-edf9-4a1f-b0b0-44c746de0e03" />

**Comment by cp3hnu** [comment]
Same issue, can be reproduced in next 15.2.3

**Comment by georgewwindsor** [comment]
bump, still a problem on latest

**Comment by ipcloudlive** [comment]
```
npx @next/codemod built-in-next-font .
```
This fixed the error

**Comment by GowthamTG** [comment]
This is still a issue in next 15.3.0

**Comment by damianpetroff** [comment]
I have a hinge that, at least in my case, it could be related to company VPN.
Anyone in the same case that could resolve the issue by disabling / working around the VPN/Firewall ?

**Reply by damianpetroff** [reply]
At least in my case, and I have the same log as OP, it really looks like a network issue while trying to fetch https://fonts.gstatic.com/*

**Reply by thhomas** [reply]
Yes same here, I have a corporate network using Zscaler and I had to switch to a free one to make it work. 
It could be because turbopack is a Rust app and fails to use my specific Root CA Certificate but I could not find a way go through it.

**Reply by Hexi1997** [reply]
same

**Reply by aaronguostudio** [reply]
Same here, got the same issue with Zscaler

**Reply by josuefuentesdev** [reply]
In my case, using the Cloudflare VPN solved my issue. It might be related to network censorship in Spain.

**Comment by aixegorri** [comment]
Using Cloudflare Warp solves the problem. I can't say for sure that the reason is censorship in Spain due to LALIGA.

**Comment by iAmAy0mide** [comment]
This appears to be a compatibility issue between **next/font** and **Turbopack**. To resolve it, update the dev script in your **package.json** by removing the **--turbopack** flag:

**Before:**
"scripts": {
  "dev": "next dev --turbopack",
  "build": "next build",
  "start": "next start",
  "lint": "next lint"
}

**After:** 
"scripts": {
  "dev": "next dev",
  "build": "next build",
  "start": "next start",
  "lint": "next lint"
}

Simply removing --turbopack resolved the issue in my case. I'm currently using Next.js version 15.3.3.

**Comment by Stanjae** [comment]
Yeah, it is related to Turbo, and the Google font Inter is used in the sample Turbo code.
Just restart your dev server, or you can temporarily turn off the turbopack feature when running the application on the dev by removing --turbopack in package.json

**Comment by luchillo17** [comment]
I installed it locally through npm though Idk if it opts me out from font optimizations: https://github.com/vercel/next.js/issues/77861#issuecomment-3017723912

**Comment by timneutkens** [comment]
Can you give next@canary a try to check if this is fixed already?

**Reply by andrebrito16** [reply]
I tried it now and got this error.

Also in my build always I got something like the image bellow. Maybe it's related to fonts as well.

![CleanShot 2025-07-04 at 16 38 13@2x](https://github.com/user-attachments/assets/06edc04f-2a64-4801-9091-462611add11e)

**Comment by tranlap2412** [comment]
upgrade next  "next": "^15.3.5" and remove  --turbopack

**Reply by luchillo17** [reply]
Removing `--turbopack` would disable some performance no?
I would rather install the fonts from NPM directly.

**Comment by oOMrCookieOo** [comment]
https://stackoverflow.com/questions/79458011/turbopack-issue-with-nextjs15-in-windows

This error is often caused by corrupted native binaries. I fix it like this:

# Clean npm cache

npm cache clean --force

# Delete node_modules and .next

rm -rf node_modules .next

# Reinstall dependencies

npm install

this seems to work for me as mention in link
even when i removed --turbopack it still bugged when i was trying to build but the cache clear did it

**Reply by nada-aldubaie2** [reply]
**In fact, this solution is temporary. As soon as you reopen the vscode again, the problem often returns.**

**Reply by oOMrCookieOo** [reply]
Sorry, that did not happen to me so i am not sure what else to do, Good luck tho!

**Comment by basicButtons** [comment]
I create the repo by 'npx create-next-app@latest', and I run 'npm run dev'. I get this error, but I fix this by doing the following things.

1. delete 'node_modules' and 'package-lock.json' files.
2. run  'pnpm i' 
3. run 'npm run dev'

**Reply by basicButtons** [reply]
It seems that this is something wrong with next package'version, so reinstall the node_modules is very intuitive.

**Comment by tonyabracadabra** [comment]
Why the problem persist till now?

**Comment by stomayid** [comment]
By me it was solved by disabling the **ad blocker** and **web filter** (VPN).

**Reply by RiverCui** [reply]
thank u, it worked.

**Comment by Mo-Ali7375** [comment]
how I solved it:
### 1- Installed geist.
### 2- Modified my layout.tsx file like this:
      import type { Metadata } from "next";
      import { GeistSans } from "geist/font/sans";
      import { GeistMono } from "geist/font/mono";
      import "./globals.css";
      
      export const metadata: Metadata = {
        title: "Create Next App",
        description: "Generated by create next app",
      };
      
      export default function RootLayout({
        children,
      }: {
        children: React.ReactNode;
      }) {
        return (
          <html lang="en">
            <body
              className={`${GeistSans.variable} ${GeistMono.variable} antialiased`}
            >
              {children}
            </body>
          </html>
        );
      }

I hope this helps you :)

**Comment by narendram224** [comment]
I got the same issue.
For me when I have added the this code in my .env.local it works.

For **Next js 16.**

`NEXT_TURBOPACK_EXPERIMENTAL_USE_SYSTEM_TLS_CERTS=1`
<img width="628" height="189" alt="Screenshot 2025-11-09 at 5 47 29 PM" src="https://github.com/user-attachments/assets/09b1689f-848d-4483-99f2-5df4859c187e" />

**Comment by 0xmanny** [comment]
updating to next 15.5.6 worked for me

**Comment by YarosMallorca** [comment]
Adding `turbopackUseSystemTlsCerts: true` to `nextConfig.config.ts` fixed it for me.

<img width="400"  alt="image" src="https://github.com/user-attachments/assets/a8a5d97f-c71c-47a3-bf5c-5f3c7c8348be" />

**Reply by Hexi1997** [reply]
work for me

**Reply by juliocesaroliveira** [reply]
thank you, it worked.

**Comment by Ammar1999y** [comment]
For me, it only works when I enable a VPN before starting the server.

**Reply by YusufcanY** [reply]
That was the cause for me. Thanks

**Comment by afpro** [comment]
for latest next.js version, remove '--turbopack' already can't fix this bug, because next.js use turbopack as default.

explicit change '--turbopack' to '--webpack' fix this issue for me (for now).

**Comment by Smoke3785** [comment]
In my case, the issue was caused when I capitalized the file extension on one of my fonts. I had: 
```tsx
const h1FontFamily = localFont({
  src: [
    {
      path: "../assets/font/AgencyFB/AGENCYB.TTF",
      weight: "400",
      style: "normal",
    },
  ],
});
```
and it threw the above error. Changing my code to:
```tsx
const h1FontFamily = localFont({
  src: [
    {
      path: "../assets/font/AgencyFB/AGENCYB.ttf",
      weight: "400",
      style: "normal",
    },
  ],
});
``` 
resolved the error and allowed me to import my font.

I also changed the file extension to lowercase, too.
